system/hardware/tici/agnos.py

```diff
@@ -83,8 +91,11 @@ def unsparsify(f: StreamingDecompressor) -> Generator[bytes, None, None]:
 
 # noop wrapper with same API as unsparsify() for non sparse images
 def noop(f: StreamingDecompressor) -> Generator[bytes, None, None]:
-  while not f.eof:
-    yield f.read(1024 * 1024)
+  while True:
+    chunk = f.read(1024 * 1024)
+    if not chunk:
```

Comment:
I think a `len` check is more readable here

Reply:
did that first and then changed my mind